src/main/java/com/google/alpollo/model/SongEntity.java

```diff
@@ -53,7 +59,7 @@ public boolean equals(Object obj) {
   } 
 
     SongEntity entity = (SongEntity) obj;
-    return salience == entity.salience
+    return DoubleMath.fuzzyEquals(salience, entity.salience, TOLERANCE)
```

Comment:
This breaks the contract between equals and hashCode, as with this two entities can be equal but have different hash codes

Reply:
Here you can use `Double.compare()` instead...

Reply:
I don't really understand how you want to use `Double.compare()` here.

Reply:
`Double.compare(salience, entity.salience)` will return 0 if the numbers are numerically equal.